core/spec/models/spree/payment_spec.rb

```diff
@@ -1231,26 +1246,17 @@
     end
   end
 
-  describe "is_cvv_risky?" do
-    it "returns false if cvv_response_code == 'M'" do
-      payment.update_attribute(:cvv_response_code, "M")
-      expect(payment.is_cvv_risky?).to eq(false)
-    end
-
-    it "returns false if cvv_response_code == nil" do
-      payment.update_attribute(:cvv_response_code, nil)
-      expect(payment.is_cvv_risky?).to eq(false)
-    end
-
-    it "returns false if cvv_response_message == ''" do
-      payment.update_attribute(:cvv_response_message, '')
-      expect(payment.is_cvv_risky?).to eq(false)
+  describe "#is_cvv_risky?" do
+    it "returns false if cvv_response_code is 'M' or nil" do
+      ['M', nil].each do |char|
+        payment.cvv_response_code = char
+        expect(payment.is_cvv_risky?).to eq(false)
+      end
     end
 
-    it "returns true if cvv_response_code == [A-Z], omitting D" do
-      # should use cvv_response_code helper
-      (%w{N P S U} << "").each do |char|
-        payment.update_attribute(:cvv_response_code, char)
+    it "returns true if cvv_response_code is not 'M' or nil" do
+      ['', *('A'...'M'), *('N'..'Z')].each do |char|
+        payment.cvv_response_code = char
         expect(payment.is_cvv_risky?).to eq(true)
       end
```

Comment:
Nit: Similar to https://github.com/solidusio/solidus/pull/4883/files#r1094171649, I'd prefer separate examples for empty string and non-M alpha characters. For the non-M scenario, I'd usually just choose one character and call it a day :D